### ISSUE TITLE
posix function used without checking if posix is available

### ISSUE THREAD

**Issue by bjoern86** [salesagility/SuiteCRM#6349]
grep -lR "posix" *

returns the following six files which are using posix

1. include/HTMLPurifier/HTMLPurifier.standalone.php 					
2. include/utils.php
3. modules/UpgradeWizard/systemCheck.php
4. modules/UpgradeWizard/uw_ajax.php
5. modules/UpgradeWizard/uw_utils.php
6. vendor/ezyang/htmlpurifier/library/HTMLPurifier/DefinitionCache/Serializer.php

the problem is, that (apart from file no. 1 and 6) the files no. 2, 3, 4 and 5 are using posix functions without checking if posix is available:

1. OK: masked propperly with "if (function_exists('posix_getuid'))"
2. NOT OK: needs fix in line 454, reffering to issue #6343, fixed with commit 32873d6ed30d3c59785fdb9b915c65d15e0aa002
3. NOT OK: needs fix in line 101 and 102
4. NOT OK: needs fix in line 895 and 896
5. NOT OK: needs fix in line 2652 and 2653
6. OK: masked propperly with "if (function_exists('posix_getuid'))"

**Comment by bjoern86** [comment]
the solution should be to change:

$owner = posix_getpwuid(fileowner($file));
$group = posix_getgrgid(filegroup($file));

to 

$owner = (function_exists('posix_getpwuid') ? posix_getpwuid(fileowner($file)) : array('name' => 'NULL'));
$group = (function_exists('posix_getgrgid') ? posix_getgrgid(filegroup($file)) : array('name' => 'NULL'));

**Comment by bjoern86** [comment]
Fixed in 7.11.6

**Comment by pgorod** [comment]
@bjoern86 thanks for all your help with this issue, we really appreciate it.
